app/earn/[...vault]/form/form-input/form-input.tsx

```diff
@@ -24,10 +25,12 @@ interface FormInputProps {
 }
 
 const FormInput: React.FC<FormInputProps> = ({ disabled }) => {
-  const { displayValue, placeholderDisplayValue, onValueChange, vault } = useVaultContext()
-  const { walletBalanceBN } = useAppSelector(state => state.vaultUser)
+  const { displayValue, placeholderDisplayValue, onValueChange, vault, formAction } = useVaultContext()
+  const { walletBalanceBN, vaultBalanceBN } = useAppSelector(state => state.vaultUser)
   const { status } = useWalletWrapperContext()
 
+  const balanceBN = formAction === FormAction.DEPOSIT ? walletBalanceBN : vaultBalanceBN
```

Comment:
It incorrect logic, wallet should allways display only wallet balance, not need to switch it. With your changes when I switch from deposit to withdraw wallet balance incorrect:
![image](https://github.com/user-attachments/assets/04f5923e-9db8-4842-a945-73ffa7df4660)
![image](https://github.com/user-attachments/assets/79fc1121-bb81-484c-872e-bc402df283c1)